golem_sci/implementation.py

```diff
@@ -477,11 +486,18 @@ def _monitor_blockchain(self):
     def _monitor_blockchain_single(self):
         if not self._update_block_numbers():
             return
-        self._update_gas_price()
-        self._pull_subscription_events()
-        self._pull_eth_subscription_events()
-        self._process_awaiting_transactions()
-        self._process_sent_transactions()
+        logger.debug("SCI monitor: block updated")
+        steps = (
+            self._update_gas_price,
+            self._pull_subscription_events,
+            self._pull_eth_subscription_events,
+            self._process_awaiting_transactions,
+            self._process_sent_transactions,
+        )
+        for step in steps:
+            if self._monitor_started:
```

Comment:
`if not started: break`?

Reply:
hmm... well, given that's it's just one line (not counting the debug), I'd say the difference is really miniscule